unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/EmotesHUD/EmotesHUDController.cs

```diff
@@ -1,69 +1,415 @@
+
+            shortcut7InputAction = Resources.Load<InputAction_Trigger>("ToggleEmoteShortcut7");
+            shortcut7InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            shortcut8InputAction = Resources.Load<InputAction_Trigger>("ToggleEmoteShortcut8");
+            shortcut8InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
 
-    private void OnViewClosed() { emotesVisible.Set(false); }
-    private void OnAvatarEmoteSet(string id, long timestamp) { emotesVisible.Set(false); }
-    private void OnCloseWindowPressed(DCLAction_Trigger action) { emotesVisible.Set(false); }
+            shortcut9InputAction = Resources.Load<InputAction_Trigger>("ToggleEmoteShortcut9");
+            shortcut9InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            auxShortcut0InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut0");
+            auxShortcut0InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            auxShortcut1InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut1");
+            auxShortcut1InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut2InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut2");
+            auxShortcut2InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut3InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut3");
+            auxShortcut3InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+
+            auxShortcut4InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut4");
+            auxShortcut4InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut5InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut5");
+            auxShortcut5InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut6InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut6");
+            auxShortcut6InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut7InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut7");
+            auxShortcut7InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut8InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut8");
+            auxShortcut8InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+            
+            auxShortcut9InputAction = Resources.Load<InputAction_Trigger>("ToggleShortcut9");
+            auxShortcut9InputAction.OnTriggered += OnNumericShortcutInputActionTriggered;
+        }
+
+        private void OnNumericShortcutInputActionTriggered(DCLAction_Trigger action)
+        {
+            if (!shortcutsCanBeUsed)
+                return;
+
+            switch (action)
```

Comment:
This switch could also be replaced by `PlayEmote(emotesCustomizationDataStore.equippedEmotes[(int)action]?.id);` if we are sure every enum value corresponds to their "ToggleEmoteShortcutXX"

Reply:
I think no because `(int)action` will return us the number assigned into the `DCLAction_Trigger` enum (142, 143, 144...) that is not what we need here... here we need the number of the pressed key (0, 1, 2... 9).